ext/intl/tests/grapheme_empty.phpt

```diff
@@ -0,0 +1,22 @@
+--TEST--
+Test grapheme_strpos-alike functions with empty needle
+--SKIPIF--
+<?php if( !extension_loaded( 'intl' ) ) print 'skip'; ?>
+--FILE--
+<?php
+
+var_dump(grapheme_strpos("abc", ""));
+var_dump(grapheme_stripos("abc", ""));
+var_dump(grapheme_strrpos("abc", ""));
+var_dump(grapheme_strripos("abc", ""));
+var_dump(grapheme_strstr("abc", ""));
+var_dump(grapheme_stristr("abc", ""));
+
+?>
+--EXPECT--
+int(0)
+int(0)
+int(3)
+int(3)
+string(3) "abc"
+bool(false)
```

Comment:
Shouldn't this one also be `string(3) "abc"`?

Reply:
Ah, yes.. I thought that the removed check from `strstr_common_handler()` was inside `if ( !f_ignore_case ) {}`, thus it only applies to `grapheme_strstr`. And when I double-checked it on 3v4n, the `intl.error_level` tricked me, since there is no error shown by default.

That said, I'll try to fix this.

Reply:
The reason of the `false` return value is this check in `grapheme_strpos_utf16()`:

```
src = usearch_open(uneedle, uneedle_len, uhaystack, uhaystack_len, "", bi, &status);
STRPOS_CHECK_STATUS(status, "Error creating search object");
```
To make things worse, `grapheme_strpos_utf16()` is called in the other grapheme string functions when the offset is below 0, so e.g. `grapheme_strpos("abc", "", -1)` also fails currently.

I don't see any solution for the problem because `usearch_open()` is third-party code, otherwise we could just add a similar check for empty needle what George added to `zend_memnstr()`. @nikic Do you know about any good solution that circumvents this problem? Is it a very big problem if `grapheme_()` functions can't handle empty needles?

Reply:
Looks like the relevant icu code is: https://github.com/unicode-org/icu/blob/f744742e036441cbfd54b535e722dc3801ff0f63/icu4c/source/i18n/usearch.cpp#L2675

I think we should add an explicit check for this case then. I do think it would be good to support empty strings here, as we also added support for them in ext/standard and ext/mbstring (I'm not sure about ext/iconv right now).

Reply:
I've reverted the change in question, but added a task (https://github.com/php/php-src/pull/5972#discussion_r481912754).